app/src/main/java/org/oppia/app/utility/ClickableAreasImage.kt

```diff
@@ -0,0 +1,82 @@
+package org.oppia.app.utility
+
+import android.graphics.RectF
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.ImageView
+import com.github.chrisbanes.photoview.OnPhotoTapListener
+import com.github.chrisbanes.photoview.PhotoViewAttacher
+import org.oppia.app.model.ImageWithRegions.LabeledRegion
+import kotlin.math.roundToInt
+
+class ClickableAreasImage(
```

Comment:
I wonder if this would be a bit cleaner if framed as a custom image view rather than a component that sits alongside the view. At the moment, there are a lot of circular references being passed around between this object & the view, the internal listener, and the external listener that it's a bit hard to verify there aren't memory leaks possible here. I think if you can make this a custom view, it might be easier to manage: it becomes plug & play, maybe with the listener & regions attached via data-binding, & it otherwise self-manages the clickable behaviors.